### PR TITLE
[KS-136] Disallow non-trigger steps with no dependents

### PR DESCRIPTION
**Pull request by cedric-cordenier**

### CODE REVIEW

File: core/services/workflows/models_test.go [hunk 1]
```diff
@@ -186,6 +186,31 @@ targets:
 				"a-target": {},
 			},
 		},
+		{
+			name: "non-trigger step with no refs",
```
Comment (bolekk) [inline core/services/workflows/models_test.go:190]: which non-trigger step has no ref here?

File: core/services/workflows/models_test.go [hunk 2]
```diff
@@ -186,6 +186,31 @@ targets:
...
+			yaml: `
+triggers:
+  - type: "a-trigger"
+  - type: "a-second-trigger"
+actions:
+  - type: "an-action"
+    ref: "an-action"
+    inputs:
+      hello: "world"
```
Comment (cedric-cordenier) [inline core/services/workflows/models_test.go:199]: @bolekk This step; ref = reference to another step

Comment (cedric-cordenier) [reply core/services/workflows/models_test.go:199]: (BTW, I realise this is super confusing; have tightened up the test and the error that we throw to be clearer)

Comment (bolekk) [reply core/services/workflows/models_test.go:199]: thanks :)

**Review by bolekk — APPROVED** [review]

**Pull request merged**